### PR TITLE
fix datasource provider

### PR DESCRIPTION
**Pull request by andyroyle**
I'm not sure if this provider has ever worked or just got broken in a previous merge, but i found several issues with it.

- missing and malformed arguments to `send_request` method
- methods `default` do not match the parameter name `is_default`

### CODE REVIEW

**Comment by andyroyle** [comment]
I'm not sure why rubocop is failing, i've not made any change to that failing file (`spec/classes/grafana_spec.rb`). It could have been caused by rubocop being updated and the defaults being changed.

I can't get this to fail on my local machine either 😢

File: .rubocop.yml [hunk 1]
```diff
@@ -430,6 +430,8 @@ Style/PerlBackrefs:
 
 Style/PredicateName:
   Enabled: True
+  Exclude:
```
Comment (bastelfreak) [inline .rubocop.yml]: please remove this. the file is managed by modulesync

**Comment by bastelfreak** [comment]
Hi @andyroyle, we currently have another PR open which will fix the rubocop issues. Please rebase after #7 got merged.

File: .rubocop.yml (hunk 1 above)
Comment (dhoppe) [inline .rubocop.yml]: We should not exclude the whole file, but you can disable a specific cop. Just take a look at the end of this documentation.
https://github.com/bbatsov/rubocop/blob/master/manual/configuration.md

**Comment by simonpasquier** [comment]
FWIW I'm the original author of the grafana_datasource and grafana_dashboard provides when they were added to @bfraser repository. They worked fine at that time. This [commit](https://github.com/voxpupuli/puppet-grafana/commit/5a12d93e0d64adea9d37b78c983d22c33a58fa12#diff-5b9a3ae3bfb0a4863d78764edfb6396f) introduced the breakage.

**Comment by doomnuggets** [comment]
This would be a great opportunity to make the API endpoint configurable.
I ran into problems when using the module with a reverse proxy setup (apache).
I'd love to create a pull request as soon as this one is merged :)

**Pull request merged**